lib/ansible/modules/net_tools/gandi_livedns.py

```diff
@@ -0,0 +1,212 @@
+
+- name: Create a mail CNAME record to www.my.com domain
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    values:
+    - www
+    api_key: dummyapitoken
+    state: present
+
+- name: Change its TTL
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    values:
+    - www
+    ttl: 10800
+    api_key: dummyapitoken
+    state: present
+
+- name: Delete the record
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    api_key: dummyapitoken
+    state: absent
+'''
+
+RETURN = r'''
+record:
+    description: A dictionary containing the record data.
+    returned: success, except on record deletion
+    type: complex
+    contains:
+        values:
+            description: The record content (details depend on record type).
+            returned: success
+            type: list
+            sample:
+            - 192.0.2.91
+            - 192.0.2.92
+        name:
+            description: The record name.
+            returned: success
+            type: str
+            sample: www
+        ttl:
+            description: The time-to-live for the record.
+            returned: success
+            type: int
+            sample: 300
+        type:
+            description: The record type.
+            returned: success
+            type: str
+            sample: A
+        domain:
+            description: The domain associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+        zone:
+            description: The zone associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.gandi_livedns_api import GandiLiveDNSAPI
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_key=dict(type='str', required=True, no_log=True),
+            record=dict(type='str', default='@', aliases=['name']),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            ttl=dict(type='int', default=None),
+            type=dict(type='str', choices=['A', 'AAAA', 'ALIAS', 'CAA', 'CDS', 'CNAME', 'DNAME', 'DS', 'KEY', 'LOC', 'MX', 'NS', 'PTR', 'SPF', 'SRV',
+                                           'SSHFP', 'TLSA', 'TXT', 'WKS']),
+            values=dict(type='list', aliases=['content']),
+            zone=dict(type='str'),
+            domain=dict(type='str'),
+        ),
+        supports_check_mode=True,
+        required_if=[
+            ('state', 'present', ['record', 'type', 'values']),
+            ('state', 'absent', ['record', 'type']),
+        ],
+    )
+
+    if not module.params['zone'] and not module.params['domain']:
+        module.fail_json(msg="At least one of zone and domain parameters need to be defined.")
+    if module.params['state'] == 'present' and module.params['ttl'] is None:
+        module.params['ttl'] = 10800
```

Comment:
is there a reason why you don't default this in the arg_spec?

Reply:
1. at least one of zone or domain is required, how could I define this in arg_spec?

2. ttl can be used as a filter when state is absent, and may be None if the user doesn't want to pass it explicitly

- {name: foo, type: A, state: absent} will remove foo A entry regardless the value of ttl
- {name: foo, type: A, ttl: 3600, state: absent} will remove foo A entry only if its ttl is 3600

So ttl is defaulted to something (10800) only if state is present.
Is there a way to have a default value in arg_spec only when state is present?